spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -1681,14 +1681,17 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
     }
   }
 
-  test("Year") {
+  test("DatePart functions: Year/Month/DayOfMonth/DayOfWeek/DayOfYear/WeekOfYear/Quarter") {
     Seq(false, true).foreach { dictionary =>
       withSQLConf("parquet.enable.dictionary" -> dictionary.toString) {
         val table = "test"
         withTable(table) {
           sql(s"create table $table(col timestamp) using parquet")
           sql(s"insert into $table values (now()), (null)")
-          checkSparkAnswerAndOperator(s"SELECT year(col) FROM $table")
+          // TODO: weekday(col) https://github.com/apache/datafusion-comet/issues/2330
+          checkSparkAnswerAndOperator(
```

Comment:
Can we change the session timezone when we read the values back? Also, maybe add some test values corresponding to days before the Unix Epoch.

Reply:
> Can we change the session timezone when we read the values back? 

The input of these methods is `DateType` and output is `IntegerType`, , so it doesn’t seem to be related to the timezone.

> Also, maybe add some test values corresponding to days before the Unix Epoch.

Thanks, I will try it

Reply:
> Also, maybe add some test values corresponding to days before the Unix Epoch.

Thanks, added

Reply:
> > Can we change the session timezone when we read the values back?
> 
> The input of these methods is `DateType` and output is `IntegerType`, , so it doesn’t seem to be related to the timezone.
> 
> > Also, maybe add some test values corresponding to days before the Unix Epoch.
> 
> Thanks, I will try it

Oh, right. Never understood why Spark/Parquet DateType have no timezone